### PR TITLE
Remove superfluous app ledger whitelists

### PR DESCRIPTION
**Pull request by algomaxj**
Now that foreign apps must be specified by an offset into `txn.ForeignApps`, these whitelists can be removed completely (the address whitelist was already redundant)

### CODE REVIEW

File: ledger/applications.go [hunk 1]
```diff
@@ -110,33 +94,16 @@ func newAppLedger(balances transactions.Balances, acctWhitelist []basics.Address
...
 }
 
 // MakeDebugAppLedger returns logic.LedgerForLogic suitable for debug or dryrun
-func MakeDebugAppLedger(balances transactions.Balances, acctWhitelist []basics.Address, appGlobalWhitelist []basics.AppIndex, appIdx basics.AppIndex, schemas basics.StateSchemas, globals AppTealGlobals) (al logic.LedgerForLogic, err error) {
-	return newAppLedger(balances, acctWhitelist, appGlobalWhitelist, appIdx, schemas, globals)
+func MakeDebugAppLedger(balances transactions.Balances, appIdx basics.AppIndex, schemas basics.StateSchemas, globals AppTealGlobals) (al logic.LedgerForLogic, err error) {
+	return newAppLedger(balances, appIdx, schemas, globals)
 }
 
 func (al *appLedger) Balance(addr basics.Address) (res basics.MicroAlgos, err error) {
```
Comment (derbear) [inline ledger/applications.go:106]: Would be very marginally more clear to get rid of these named returned values here.

File: ledger/applications.go [hunk 2]
```diff
@@ -227,12 +184,6 @@ func (al *appLedger) AppLocalState(addr basics.Address, appIdx basics.AppIndex)
 }
 
 func (al *appLedger) AssetHolding(addr basics.Address, assetIdx basics.AssetIndex) (holding basics.AssetHolding, err error) {
```
Comment (derbear) [inline ledger/applications.go:186]: (see above on named return values)

File: ledger/applications.go [hunk 3]
```diff
@@ -250,12 +201,6 @@ func (al *appLedger) AssetHolding(addr basics.Address, assetIdx basics.AssetInde
 }
 
 func (al *appLedger) AssetParams(addr basics.Address, assetIdx basics.AssetIndex) (params basics.AssetParams, err error) {
```
Comment (derbear) [inline ledger/applications.go:203]: (see above on named return values)

File: ledger/applications.go (hunk 1 above)
Comment (algomaxj) [reply ledger/applications.go:106]: done!

File: ledger/applications.go (hunk 2 above)
Comment (algomaxj) [reply ledger/applications.go]: done!

File: ledger/applications.go (hunk 3 above)
Comment (algomaxj) [reply ledger/applications.go]: done!

**Pull request merged**